manager/resource.go

```diff
@@ -85,7 +85,7 @@ func (rm *ResourceManager) SetPods(pods *v1.PodList) {
 	for k, p := range pods.Items {
 		rm.pods[rm.getStoreKey(p.Namespace, p.Name)] = &pods.Items[k]
 
-		rm.incrementRefCounters(&p)
+		rm.incrementRefCounters(&pods.Items[k])
```

Comment:
Instead of looking the item back up, can we re-scope `p` in the loop?... e.g. `p := p`

Reply:
Sure, which would be:

```
for k, _ := range pods.Items {
    p := pods.Items[k]
    rm.pods[rm.getStoreKey(p.Namespace, p.Name)] = &p
    rm.incrementRefCounters(&p)
}
```

Reply:
Oh, I see. But we do observe pod deletion. I double checked it, and found the reason is we are using a forked branch, which has not been synced to the latest upstream master branch for a while. At our branch, the `func (rm *ResourceManager) SetPods(pods *v1.PodList) {` logic is like:

```
for k, p := range pods.Items {
		if p.Status.Phase == v1.PodSucceeded {
			continue
		}
		rm.pods[rm.getStoreKey(p.Namespace, p.Name)] = &pods.Items[k]
		rm.incrementRefCounters(&p)
	}
```

So, when watch is broken and trigger a relist, succeeded pods will been deleted. This has been already fixed at upstream, but we have not synced it yet.